cms/djangoapps/contentstore/views/item.py

```diff
@@ -631,10 +665,77 @@ def _create_item(request):
     )
 
     return JsonResponse(
-        {"locator": unicode(created_block.location), "courseKey": unicode(created_block.location.course_key)}
+        {'locator': unicode(created_block.location), 'courseKey': unicode(created_block.location.course_key)}
     )
 
 
+def _get_source_index(source_usage_key):
+    """
+    Get source index position of the XBlock.
+
+    Arguments:
+        source_usage_key (BlockUsageLocator): Locator of source item.
+
+    Returns:
+        source_index (int): Index position of the xblock in a parent.
+    """
+    store = modulestore()
+    source_item = store.get_item(source_usage_key)
+    source_parent = source_item.get_parent()
+    return source_parent.children.index(source_usage_key)
+
+
+def is_valid_move(source_usage_key, dest_parent_usage_key):
+    """
+    Checks if move operation is valid or not.
+
+    Arguments:
+        source_usage_key (BlockUsageLocator): Locator of source item.
+        dest_parent_usage_key (BlockUsageLocator): Locator of destination item.
+
+    Returns:
+       True if valid move action otherwise False.
+    """
+    source_type = source_usage_key.block_type
+    dest_type = dest_parent_usage_key.block_type
+
+    valid_move_children = {
+        'vertical': source_type if source_type in COMPONENT_TYPES else 'component',
+        'sequential': 'vertical',
+        'chapter': 'sequential',
+    }
+
+    return valid_move_children.get(dest_type, '') == source_type
+
+
+def _move_item(source_usage_key, dest_parent_usage_key, user, source_index=None):
+    """
+    Move an existing xblock as a child of the supplied dest_parent_usage_key
+
+    If source_index is provided, xblock would be placed under dest_parent_usage_key on that particular position.
+
+    Arguments:
+            source_usage_key (BlockUsageLocator): Locator of source item.
+            dest_usage_key (BlockUsageLocator): Locator of destination item.
+            source_index (int): If provided, insert source item at the provided index location in dest_usage_key item.
+    """
+    store = modulestore()
+    source_item = store.get_item(source_usage_key)
+    source_parent = source_item.get_parent()
+    dest_parent = store.get_item(dest_parent_usage_key)
+
+    # Remove reference from old parent
+    source_parent.children.remove(source_item.location)
```

Comment:
These changes should be made inside a transaction so that if something fails it doesn't leave the course in an inconsistent state.